### PR TITLE
Update Linux Apt Commands to Handle Lock Timeouts

### PR DESCRIPTION
**Pull request by nagilson**
resolves https://github.com/dotnet/vscode-dotnet-runtime/issues/1806#event-13429482290

A lot of users are seeing failure due to the DPKG lock being held by another process. We can add a flag to wait for the lock to be freed. This wont fix everything, as Apt will return 100 whenever there is any failure. https://github.com/Debian/apt/blob/aa56836331870d975c212a5df2f13db9ce3914bf/apt-private/private-cmndline.cc#L602

This means the errors we are seeing are not just due to locking errors. But, perhaps this will help.

This will also add a timeout to our own executions so that they dont keep the lock.

Also this will try to run update twice, to see if that 2nd try will work. It seems a bit silly, but it does actually seem to improve the chance of success when something else is running on the system.

The timeout time is mildly arbitrary.... we can try to improve that in the future to be based off a fraction of the total timeout time, or something like this. But I would like to see the impact of how much this helps without this change.

### CODE REVIEW

File: vscode-dotnet-runtime-library/distro-data/distro-support.json [hunk 1]
```diff
@@ -34,15 +43,18 @@
...
                 ]
             },
             {
                 "runUnderSudo": true,
                 "commandRoot": "apt-get",
                 "commandParts": [
+                    "-o",
+                    "DPkg::Lock::Timeout=120",
                     "upgrade",
-                    "-y",
```
Comment (nagilson) [inline vscode-dotnet-runtime-library/distro-data/distro-support.json:45]: To check: does apt-get upgrade not accept -y? need to test tmrw on a vm.

File: vscode-dotnet-runtime-library/src/Acquisition/InstallTrackerSingleton.ts [hunk 2]
```diff
@@ -77,26 +78,33 @@ export class InstallTrackerSingleton
...
-            finally
+            else
             {
-                await release();
+                await lockfile.lock(lockPath, { retries: { retries: 10, minTimeout: 5, maxTimeout: 10000 } })
+                .then(async (release) =>
+                {
+                    returnResult = await f(...(args));
+                    this.eventStream?.post(new DotnetLockReleasedEvent(`Lock about to be released.`, new Date().toISOString(), lockPath, lockPath));
+                    await release();
```
Comment (nagilson) [inline vscode-dotnet-runtime-library/src/Acquisition/InstallTrackerSingleton.ts:101]:
```suggestion
                    return release();
```

File: vscode-dotnet-runtime-library/src/Acquisition/InstallTrackerSingleton.ts [hunk 3]
```diff
@@ -77,26 +78,33 @@ export class InstallTrackerSingleton
...
-            const release = await lockfile.lock(lockPath, { retries: { retries: 10, minTimeout: 5, maxTimeout: 2000 } });
-            try
-            {
-                return await f(...(args));
-            }
-            finally
+            else
             {
-                await release();
+                await lockfile.lock(lockPath, { retries: { retries: 10, minTimeout: 5, maxTimeout: 10000 } })
```
Comment (MiYanni) [inline vscode-dotnet-runtime-library/src/Acquisition/InstallTrackerSingleton.ts:96]: What units are the min and max timeout in?

**Review by MiYanni — APPROVED** [review]

File: vscode-dotnet-runtime-library/src/Acquisition/InstallTrackerSingleton.ts (hunk 3 above)
Comment (nagilson) [reply vscode-dotnet-runtime-library/src/Acquisition/InstallTrackerSingleton.ts:96]: they are in MS. The library will do an exponential equation where it starts trying again after 5 ms and then takes longer each retry until its at 10000 ms.

File: vscode-dotnet-runtime-library/distro-data/distro-support.json (hunk 1 above)
Comment (nagilson) [reply vscode-dotnet-runtime-library/distro-data/distro-support.json:45]: it does.

**Pull request merged**